auth/auth_test.go

```diff
@@ -257,6 +258,17 @@ func TestCertificateRequestError(t *testing.T) {
 	}
 }
 
+func testNewHTTPClientNoOpts(t *testing.T) {
```

Comment:
I don't see how this test can pass without setting any env variables or downloading files. 
Should we move it to the integration tests?